src/xrpld/app/tx/detail/VaultClawback.cpp

```diff
@@ -37,17 +35,6 @@ VaultClawback::preflight(PreflightContext const& ctx)
         // Note, zero amount is valid, it means "all". It is also the default.
         if (*amount < beast::zero)
             return temBAD_AMOUNT;
-        else if (isXRP(amount->asset()))
-        {
-            JLOG(ctx.j.debug()) << "VaultClawback: cannot clawback XRP.";
-            return temMALFORMED;
-        }
```

Comment:
I think it makes sense to leave the XRP check here in `preflight`.


To summarize 3.3.1.1 from the proposed spec changes:
* If the transaction is submitted by the vault owner, then amount must be specified as the share type (MPT).
* If the transaction is submitted by the asset issuer, the amount must be specified as the asset type (IOU, MPT).

Thus, specifying `sfAmount` as XRP is _never_ valid. May as well check it here, and save the effort later.